Make questions in Q&A tab collapse to 3 lines (preview, extends when clicked).

max-krichenbauer opened the pull request:
This is an implementation of https://github.com/ros-infrastructure/rosindex/issues/197 
To make the list of questions easier to navigate, all questions are collapsed to show only the first three lines of text.
![image](https://user-images.githubusercontent.com/15894344/71011275-745fe100-20ed-11ea-90a3-4c71f835f623.png)

When clicked, the box expands to show the full text as before.
Clicked again, the box collapses again to the first three lines.

max-krichenbauer committed:
Made question descriptions collapse to 3 lines unless clicked.

gavanderhoorn commented:
Thanks for the PR @max-krichenbauer.

Overall comment: would we want to add some ellipses after the first three lines, so we give an indication of the fact there is 'more' than just what is shown?

> When clicked, the box expands to show the full text as before.
> Clicked again, the box collapses again to the first three lines.

Personally I would prefer clicks to open the ROS Answers page instead of showing them here on the index.

tfoote commented on _includes/package_body.html:
```diff
@@ -505,6 +508,11 @@ <h4 class="w-75 float-left">{{title}}</h4>
               });
+              jQuery("#{{distro}}-question-list .question-descrition").each(function() {
+                jQuery(this).click( function() {
+                  jQuery(this).toggleClass( "question-descrition-collapsed", 1000 );
```
```suggestion
                  jQuery(this).toggleClass( "question-description-collapsed", 1000 );
```

tfoote commented on css/prettify.css:
```diff
@@ -1 +1,2 @@
-.pln{color:#000}@media screen{.str{color:#080}.kwd{color:#008}.com{color:#800}.typ{color:#606}.lit{color:#066}.clo,.opn,.pun{color:#660}.tag{color:#008}.atn{color:#606}.atv{color:#080}.dec,.var{color:#606}.fun{color:red}}@media print,projection{.kwd,.tag,.typ{font-weight:700}.str{color:#060}.kwd{color:#006}.com{color:#600;font-style:italic}.typ{color:#404}.lit{color:#044}.clo,.opn,.pun{color:#440 ...
\ No newline at end of file
+.pln{color:#000}@media screen{.str{color:#080}.kwd{color:#008}.com{color:#800}.typ{color:#606}.lit{color:#066}.clo,.opn,.pun{color:#660}.tag{color:#008}.atn{color:#606}.atv{color:#080}.dec,.var{color:#606}.fun{color:red}}@media print,projection{.kwd,.tag,.typ{font-weight:700}.str{color:#060}.kwd{color:#006}.com{color:#600;font-style:italic}.typ{color:#404}.lit{color:#044}.clo,.opn,.pun{color:#440 ...
+div.question-descrition-collapsed{height: 70px;overflow: hidden;}
```
```suggestion
div.question-description-collapsed{height: 70px;overflow: hidden;}
```

max-krichenbauer replied:
Thanks! I fixed the typo.

tfoote reviewed (commented):
Thanks @max-krichenbauer you got it right from my description.

I agree that an elipses or maybe a "Show More" then "Show Less button or link would be great to add for discoverability. And changing the click on the box to go to answers.ros.org in a new window would be best. Again for discoverability a "View Question and Answers on answers.ros.org" text would be valuable too.

max-krichenbauer committed:
Fixed a typo.

max-krichenbauer committed:
Changed the Q&A display behavior according to feedback.

- a click on the collapsed text now opens the question
  on answers.ros.org in a new window/tab
- below the collapsed text is a "Show more" anchor that
  when pressed expands the text and turns into "Show less"

max-krichenbauer commented:
Thanks for the feedback. I changed it accordingly:
- a click on the collapsed text now opens the question on answers.ros.org in a new window/tab
- below the collapsed text is a "Show more" anchor that when pressed expands the text and turns into "Show less"

![image](https://user-images.githubusercontent.com/15894344/71063083-4e2d5600-216c-11ea-93e7-bc708c776dfa.png)


> Again for discoverability a "View Question and Answers on answers.ros.org" text would be valuable too.

Sorry, I don't understand what you mean.
There are already three links at the top of the list: one to general answers.ros.org, one for answers on this package and one for only the highest ranking answers on this package. 
![image](https://user-images.githubusercontent.com/15894344/71063224-9482b500-216c-11ea-843e-248dc59a878e.png)

tfoote commented:
> > Again for discoverability a "View Question and Answers on answers.ros.org" text would be valuable too.
> 
> Sorry, I don't understand what you mean.
> There are already three links at the top of the list: one to general answers.ros.org, one for answers on this package and one for only the highest ranking answers on this package.

I meant just add an anchor for each specific question so that you have a call to click somewhere and you don't have to guess that clicking on the body or heading will link you to the question. Just have some specific text next to the Show More. pseudo code here but. `<a href='...'>Show more</a> or <a href="answers.ros.org/q/XYZ">view this on answers.ros.org</a>` It's redundant with the existing links but it's discoverable and looks like a link versus the other things you need to notice the mouse changes shape before you know you can click on the content.

max-krichenbauer committed:
Added a "View on answers.ros.org" link to the end of each question.

max-krichenbauer commented:
Thank you! Now I understood what you mean.
Here it is:
![image](https://user-images.githubusercontent.com/15894344/71148851-1b976200-222d-11ea-81c4-4f6fb32f6c0c.png)
(I made it stick to the right side because the two links next to each other were not visually clear).

tfoote commented on css/prettify.css:
```diff
@@ -1 +1,4 @@
-.pln{color:#000}@media screen{.str{color:#080}.kwd{color:#008}.com{color:#800}.typ{color:#606}.lit{color:#066}.clo,.opn,.pun{color:#660}.tag{color:#008}.atn{color:#606}.atv{color:#080}.dec,.var{color:#606}.fun{color:red}}@media print,projection{.kwd,.tag,.typ{font-weight:700}.str{color:#060}.kwd{color:#006}.com{color:#600;font-style:italic}.typ{color:#404}.lit{color:#044}.clo,.opn,.pun{color:#440 ...
\ No newline at end of file
+.pln{color:#000}@media screen{.str{color:#080}.kwd{color:#008}.com{color:#800}.typ{color:#606}.lit{color:#066}.clo,.opn,.pun{color:#660}.tag{color:#008}.atn{color:#606}.atv{color:#080}.dec,.var{color:#606}.fun{color:red}}@media print,projection{.kwd,.tag,.typ{font-weight:700}.str{color:#060}.kwd{color:#006}.com{color:#600;font-style:italic}.typ{color:#404}.lit{color:#044}.clo,.opn,.pun{color:#440 ...
+div.question-description-collapsed{height: 70px;overflow: hidden;cursor: pointer;}
```
Sorry I didn't catch this earlier. This isn't the right place to add the styling. I believe that this is a standard versioned css that we don't want to modify. 

I believe that we want to drop the styling into _base.css https://github.com/ros-infrastructure/rosindex/blob/ros2/_sass/_base.scss

max-krichenbauer replied:
Thanks! I moved the CSS code to `_base.css`.

tfoote reviewed (changes requested):
Thanks that's exactly what I was thinking. And the right justification looks better. One piece of feedback on the styling location and I think we're all set.

max-krichenbauer committed:
Moved the CSS definitions from `prettify.css` to `_base.scss`.

tfoote reviewed (approved):
Thanks for the update.
